beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -7175,7 +7176,15 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                 );
                 Ok(Some(StoreOp::PutBlobs(block_root, blobs)))
             }
-            AvailableBlockData::DataColumns(data_columns) => {
+            AvailableBlockData::DataColumns(mut data_columns) => {
+                let columns_to_custody = self.custody_columns_for_epoch(Some(
+                    block_slot.epoch(T::EthSpec::slots_per_epoch()),
+                ));
+                // Supernodes need to persist all sampled custody columns
```

Comment:
```suggestion
                // If `columns_to_custody` contains all possible custody groups, no need to filter
```
I would update in the comment the mention of supernode. In the future, nodes custodying >50% of columns and doing reconstruction may be supernodes?

Reply:
I think this is fine, supernode is explicitly defined here in the spec
https://github.com/ethereum/consensus-specs/blob/dev/specs/fulu/p2p-interface.md#supernodes